modules/minimint-mint/src/lib.rs

```diff
@@ -263,43 +257,37 @@ impl FederationModule for Mint {
             .filter_map(|(issuance_id, shares)| {
                 let mut batch = DbBatch::new();
                 let mut batch_tx = batch.transaction();
+                let (bsig, errors) = self.combine(shares.clone());
 
-                if shares.len() > self.threshold {
-                    debug!(
-                        "Trying to combine sig shares for issuance request {}",
-                        issuance_id
-                    );
-                    let (bsig, errors) = self.combine(shares.clone());
-                    // FIXME: validate shares before writing to DB to make combine infallible
-                    if !errors.0.is_empty() {
-                        warn!("Peer sent faulty share: {:?}", errors);
-                    }
+                // FIXME: validate shares before writing to DB to make combine infallible
+                if !errors.0.is_empty() {
+                    warn!("Peer sent faulty share: {:?}", errors);
+                }
 
-                    match bsig {
-                        Ok(blind_signature) => {
-                            debug!(
-                                "Successfully combined signature shares for issuance request {}",
-                                issuance_id
-                            );
-
-                            batch_tx.append_from_iter(shares.into_iter().map(|(peer, _)| {
-                                BatchItem::delete(ReceivedPartialSignatureKey {
-                                    request_id: issuance_id,
-                                    peer_id: peer,
-                                })
-                            }));
-
-                            batch_tx.append_insert(OutputOutcomeKey(issuance_id), blind_signature);
-                            batch_tx.commit();
-                            Some(batch)
-                        }
-                        Err(e) => {
+                match bsig {
+                    Ok(blind_signature) => {
+                        debug!(
+                            "Successfully combined signature shares for issuance request {}",
+                            issuance_id
+                        );
+
+                        batch_tx.append_from_iter(shares.into_iter().map(|(peer, _)| {
+                            BatchItem::delete(ReceivedPartialSignatureKey {
+                                request_id: issuance_id,
+                                peer_id: peer,
+                            })
+                        }));
+
+                        batch_tx.append_insert(OutputOutcomeKey(issuance_id), blind_signature);
+                        batch_tx.commit();
+                        Some(batch)
+                    }
+                    Err(e) => {
+                        if e != CombineError::TooFewShares {
```

Comment:
Maybe make the other case a trace after removing the debug output `"Trying to combine sig shares for issuance request {}",`